src/platform/riot/riot.c

```diff
@@ -64,12 +64,12 @@ lf_ret_t PlatformRiot_wait_for(Platform *self, interval_t duration) {
 
 void PlatformRiot_leave_critical_section(Platform *self) {
   PlatformRiot *p = (PlatformRiot *)self;
-  p->irq_mask = irq_disable();
+  irq_restore(p->irq_mask);
```

Comment:
That's a great catch.  Leaving the critical section means enabling IRQs. A little surprised this did not cause any issues earlier...